internal/client/cloud_client.go

```diff
@@ -91,8 +91,52 @@ func (c *CloudClient) Do(req *http.Request) (*ResponseWithETag, error) {
 		return nil, err
 	}
 
-	// Extract the ETag if it exists
-	etag := resp.Header.Get("ETag")
+	// Extract the ETag - check multiple possible header names
+	var etag string
+
+	// Check standard ETag header (case-insensitive)
+	etag = resp.Header.Get("ETag")
+
+	// If not found, check for other variations
+	if etag == "" {
+		etag = resp.Header.Get("etag")
+	}
+	if etag == "" {
+		etag = resp.Header.Get("Etag")
+	}
```

Comment:
Did you observe that case mattered? Headers are generally assumed to be case-insensitive, and it looks like the [Header.Get](https://pkg.go.dev/net/http#Header.Get) implementation respects that.